# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.cpp
# Commit: 5aedde81baa3
# Full Hash: 5aedde81baa37b849c0868186498ef14cc2a0f92
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-06-10 04:36:07
# Description:
#   Bug 1644266 - Check whether inner window has extant document before using r=timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78872
# ==============================================================================

diff -r d6ffedc68f99 -r 5aedde81baa3 toolkit/components/antitracking/ContentBlocking.cpp
--- a/toolkit/components/antitracking/ContentBlocking.cpp	Tue Jun 09 13:46:54 2020 +0200
+++ b/toolkit/components/antitracking/ContentBlocking.cpp	Tue Jun 09 10:51:11 2020 +0000
@@ -576,12 +576,17 @@
       AntiTrackingUtils::GetInnerWindow(aParentContext);
   MOZ_ASSERT(parentInner);
 
+  Document* doc = parentInner->GetExtantDoc();
+  if (NS_WARN_IF(!doc)) {
+    return;
+  }
+
   // Theoratically this can be done in the parent process. But right now,
   // we need the channel while notifying content blocking events, and
   // we don't have a trivial way to obtain the channel in the parent
   // via BrowsingContext. So we just ask the child to do the work.
   ContentBlockingNotifier::OnEvent(
-      parentInner->GetExtantDoc()->GetChannel(), false,
+      doc->GetChannel(), false,
       CookieJarSettings::IsRejectThirdPartyWithExceptions(aCookieBehavior)
           ? nsIWebProgressListener::STATE_COOKIES_BLOCKED_FOREIGN
           : nsIWebProgressListener::STATE_COOKIES_BLOCKED_TRACKER,
